src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -154,7 +154,10 @@ public void Dispose()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
+                if (_solutionEvents != null)
```

Comment:
It should never fail, since this is removing the event handler, if solutionrestoreworker is constructed but never initialized and called into dispose then this code change makes sense. It's being defensive against some random null exception